File: 12_Bit_Manipulation/461_Hamming_Distance.cpp
# Facebook
class Solution {
public:
    int hammingDistance(int x, int y) {
        int t = x ^ y;
        int ans = 0;
        while(t>0){
            ans += t & 1; // check lowest bit
            t >>= 1;
        }
        return ans;
    }
};

File: 12_Bit_Manipulation/461_Hamming_Distance.py
""" L1: bit-manipulation
"""
class Solution:
    def hammingDistance(self, x: int, y: int) -> int:
        diff = x ^ y
        ans = 0
        while diff:
            ans += diff ^ 1
            diff >>= 1
        return ans

# pythonic
class Solution:
    def hammingDistance(self, x, y):
        return bin(x^y).count('1')
